src/Microsoft.Health.Dicom.Functions/BlobMigration/CleanupDeletedDurableFunction.Orchestration.cs

```diff
@@ -0,0 +1,110 @@
+    /// </summary>
+    /// <param name="context">The context for the orchestration instance.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>A task representing the <see cref="CleanupDeletedFilesAsync"/> operation.</returns>
+    [FunctionName(nameof(CleanupDeletedFilesAsync))]
+    public async Task CleanupDeletedFilesAsync(
+        [OrchestrationTrigger] IDurableOrchestrationContext context,
+        ILogger logger)
+    {
+        // The ID should be a GUID as generated by the trigger, but we'll assert here just to make sure!
+        EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
+        BlobMigrationCheckpoint input = context.GetInput<BlobMigrationCheckpoint>();
+        EnsureArg.IsNotNull(input, nameof(input));
+        EnsureArg.IsNotNull(input.Batching, nameof(input.Batching));
+
+        logger = context.CreateReplaySafeLogger(logger);
+
+        // Get the max watermark of all deleted instances for the exit scenario
+        long maxWatermark = input.MaxWatermark ?? await context.CallActivityWithRetryAsync<long>(
+                 nameof(GetMaxDeletedChangeFeedWatermarkAsync),
+                 _options.RetryOptions,
+                 new CleanupDeletedBatchArguments
+                 {
+                     FilterTimeStamp = input.FilterTimeStamp
+                 });
+
+        IReadOnlyList<ChangeFeedEntry> changeFeedEntries = await context.CallActivityWithRetryAsync<IReadOnlyList<ChangeFeedEntry>>(
+            nameof(GetDeletedChangeFeedInstanceBatchesAsync),
+            _options.RetryOptions,
+            new CleanupDeletedBatchArguments
+            {
+                StartWatermark = input.Completed.HasValue ? input.Completed.Value.Start : default,
+                EndWatermark = input.Completed.HasValue ? input.Completed.Value.End : default,
+                BatchSize = input.Batching.Size,
+                FilterTimeStamp = input.FilterTimeStamp == DateTime.MinValue ? null : input.FilterTimeStamp,
+            });
+
+        bool hasReachedToEnd = input.Completed.HasValue && input.Completed.Value.End >= maxWatermark;
+
+        if (changeFeedEntries.Count > 0)
+        {
+            var batchRange = new WatermarkRange(changeFeedEntries[0].Sequence, changeFeedEntries[^1].Sequence);
+
+            logger.LogInformation("Beginning to delete the range {Range}.", batchRange);
+
+            var instanceIdentifiers = changeFeedEntries.Select(x => new VersionedInstanceIdentifier(x.StudyInstanceUid, x.SeriesInstanceUid, x.SopInstanceUid, x.OriginalVersion)).ToList();
+            await context.CallActivityWithRetryAsync(nameof(CleanupDeletedBatchAsync), _options.RetryOptions, instanceIdentifiers);
+
+            // Create a new orchestration with the same instance ID to process the remaining data
+            logger.LogInformation("Completed deleting the range {Range}. Total files deleted in range: '{NumFiles}'. " +
```

Comment:
nit: technically the maximum number of files in range is the batch size, right? Not the difference?

Reply:
hmm, I don't think so, when we query using the watermark between 100-200, there can be only 50 records that are deleted. So technically it is not equal to the batch size

Reply:
That's correct, will change to Count. I was answering for the batch size property